sdk/core/Azure.Core/src/Shared/IBearerTokenChallengeOptions.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Options to be exposed in client options classes related to bearer token authorization challenge scenarios.
+    /// </summary>
+    internal interface IBearerTokenChallengeOptions
```

Comment:
This type seems to only be useful for storage can we move it into storage shared code?

Reply:
It will be used by every service that supports tenantId challenges. Storage is just the first one to implement it.

Reply:
Where would we pass an instance of `IBearerTokenChallengeOptions`?

Reply:
It's not passed so much as implemented by each ClientOptions class. In theory it could be passed to the auth policy itself rather than the value of the property.

Reply:
>In theory it could be passed to the auth policy itself rather than the value of the property.

We would make this interface non-shared for that.